### PR TITLE
feat(iot_hub): add mqtt certificate attributes

### LINKED ISSUE #2264: Provide IoT Hub MQTT CA
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I've been setting up a test environment for IoT Hub, and have managed to get so far that the only thing I'm required to download via web UI is the MQTT networks CA file. I at least can't find a way to get it via the `scaleway_iot_hub` or `scaleway_iot_device` -objects.

### New or Affected Resource(s)

* scaleway_iot_hub
* scaleway_iot_device

### Potential Terraform Configuration

How I'd see it as working would be something like this:
```hcl
resource "scaleway_iot_hub" "iot-hub" {
  name = "iot-hub"
  product_plan = "plan_shared"
}

resource "local_sensitive_file" "iot-hub-ca" {
  filename = "${path.root}/iot-hub-ca.pem"
  content = scaleway_iot_hub.iot-hub.mqtt-ca
  directory_permission = 0700
  file_permission = 0400
}
```

Another option would be to add it to the device-certificates, but as it's not device-specific that doesn't seem like the correct place.

### PR DESCRIPTION
**Pull request by Laure-di**
close #2264

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -32,6 +32,8 @@ website/vendor
 # Test exclusions
 !command/test-fixtures/**/*.tfstate
 !command/test-fixtures/**/.terraform/
+scaleway/testlocal/
```
Comment (remyleone) [inline .gitignore:35]: Please remove this

File: scaleway/helpers_iot.go [hunk 2]
```diff
@@ -12,6 +16,8 @@ import (
 const (
 	defaultIoTRetryInterval = 5 * time.Second
 	defaultIoTHubTimeout    = 5 * time.Minute
+	mqttCaNameFile          = "iot-hub-ca.pem"
+	mqttCaURLDownload       = "https://iot.s3.nl-ams.scw.cloud/certificates/"
```
Comment (remyleone) [inline scaleway/helpers_iot.go:20]: Is it a problem that region is hard coded here? Would it be the same CA regardless of the region?

File: scaleway/helpers_iot.go [hunk 3]
```diff
@@ -2,6 +2,10 @@ package scaleway
 
 import (
 	"context"
+	"fmt"
+	"io"
+	"net/http"
+	url2 "net/url"
```
Comment (remyleone) [inline scaleway/helpers_iot.go:8]: why is it named url2?

File: scaleway/helpers_iot.go (hunk 2 above)
Comment (Laure-di) [reply scaleway/helpers_iot.go:20]: I check with Julien the certificat is in a s3 in ams and it won't change in the futur, so should not be an issue if it is hard coded

File: scaleway/helpers_iot.go (hunk 3 above)
Comment (Laure-di) [reply scaleway/helpers_iot.go:8]: my bad I just change it

File: scaleway/resource_iot_hub.go [hunk 4]
```diff
@@ -138,6 +149,7 @@ func resourceScalewayIotHubCreate(ctx context.Context, d *schema.ResourceData, m
 	}
 
 	res, err := iotAPI.CreateHub(req, scw.WithContext(ctx))
+	log.Println("res from create hub ", res)
```
Comment (Codelax) [inline scaleway/resource_iot_hub.go:152]: debug log

File: scaleway/resource_iot_hub.go [hunk 5]
```diff
@@ -189,6 +201,8 @@ func resourceScalewayIotHubCreate(ctx context.Context, d *schema.ResourceData, m
 			return diag.FromErr(err)
 		}
 	}
+	MQTTUrl := computeIotHubCaURL(req.ProductPlan)
```
Comment (Codelax) [inline scaleway/resource_iot_hub.go:204]: Output an error or warn on error

Comment (Codelax) [reply scaleway/resource_iot_hub.go:204]: Output an error if mqtt_ca is empty and warn if there is already a certificate stored

Comment (Codelax) [reply scaleway/resource_iot_hub.go:204]:
```suggestion
return diag.Diagnostics{{
  Severity: diag.Warning,
  Path: cty.AttrPath("mqtt_ca")
}}
```

File: scaleway/resource_iot_hub_test.go [hunk 6]
```diff
@@ -64,6 +64,8 @@ func TestAccScalewayIotHub_Minimal(t *testing.T) {
 					resource.TestCheckResourceAttrSet("scaleway_iot_hub.minimal", "endpoint"),
 					resource.TestCheckResourceAttr("scaleway_iot_hub.minimal", "device_count", "0"),
 					resource.TestCheckResourceAttr("scaleway_iot_hub.minimal", "connected_device_count", "0"),
+					resource.TestCheckResourceAttr("scaleway_iot_hub.minimal", "mqtt_ca_url", ""),
```
Comment (Codelax) [inline scaleway/resource_iot_hub_test.go:67]: Add a comment to explain why it is supposed to be empty

File: scaleway/resource_iot_hub.go [hunk 7]
```diff
@@ -226,6 +239,16 @@ func resourceScalewayIotHubRead(ctx context.Context, d *schema.ResourceData, met
...
 	_ = d.Set("events_topic_prefix", response.EventsTopicPrefix)
 	_ = d.Set("device_auto_provisioning", response.EnableDeviceAutoProvisioning)
+	_ = d.Set("mqtt_ca_url", computeIotHubCaURL(response.ProductPlan, region))
+	MqttURL := d.Get("mqtt_ca_url")
+	MqttCa, err := computeIotHubMQTTCa(ctx, fmt.Sprintf("%v", MqttURL), meta)
+	if err != nil {
+		_ = diag.Diagnostic{
+			Severity:      diag.Warning,
+			AttributePath: cty.GetAttrPath("mqtt_ca"),
+		}
```
Comment (Codelax) [inline scaleway/resource_iot_hub.go:249]: Diagnostic content is missing

File: scaleway/resource_iot_hub.go [hunk 8]
```diff
@@ -226,6 +239,16 @@ func resourceScalewayIotHubRead(ctx context.Context, d *schema.ResourceData, met
 	_ = d.Set("disable_events", response.DisableEvents)
 	_ = d.Set("events_topic_prefix", response.EventsTopicPrefix)
 	_ = d.Set("device_auto_provisioning", response.EnableDeviceAutoProvisioning)
+	_ = d.Set("mqtt_ca_url", computeIotHubCaURL(response.ProductPlan, region))
+	MqttURL := d.Get("mqtt_ca_url")
+	MqttCa, err := computeIotHubMQTTCa(ctx, fmt.Sprintf("%v", MqttURL), meta)
```
Comment (Codelax) [inline scaleway/resource_iot_hub.go:244]:
```suggestion
	mqttCA, err := computeIotHubMQTTCa(ctx, fmt.Sprintf("%v", MqttURL), meta)
```

File: scaleway/resource_iot_hub.go [hunk 9]
```diff
@@ -226,6 +239,16 @@ func resourceScalewayIotHubRead(ctx context.Context, d *schema.ResourceData, met
 	_ = d.Set("disable_events", response.DisableEvents)
 	_ = d.Set("events_topic_prefix", response.EventsTopicPrefix)
 	_ = d.Set("device_auto_provisioning", response.EnableDeviceAutoProvisioning)
+	_ = d.Set("mqtt_ca_url", computeIotHubCaURL(response.ProductPlan, region))
+	MqttURL := d.Get("mqtt_ca_url")
```
Comment (Codelax) [inline scaleway/resource_iot_hub.go:243]:
```suggestion
	mqttURL := d.Get("mqtt_ca_url")
```

File: scaleway/resource_iot_hub.go [hunk 10]
```diff
@@ -226,6 +239,17 @@ func resourceScalewayIotHubRead(ctx context.Context, d *schema.ResourceData, met
...
 	_ = d.Set("events_topic_prefix", response.EventsTopicPrefix)
 	_ = d.Set("device_auto_provisioning", response.EnableDeviceAutoProvisioning)
+	_ = d.Set("mqtt_ca_url", computeIotHubCaURL(response.ProductPlan, region))
+	mqttURL := d.Get("mqtt_ca_url")
+	mqttCa, err := computeIotHubMQTTCa(ctx, fmt.Sprintf("%v", mqttURL), meta)
+	if err != nil {
+		_ = diag.Diagnostic{
+			Severity:      diag.Warning,
+			AttributePath: cty.GetAttrPath("mqtt_ca"),
+			Summary:       err.Error(),
```
Comment (Codelax) [inline scaleway/resource_iot_hub.go:249]: Diagnostic allows a Summary and details.
Summary could be used to summarize the issue (ex: Failed to fetch io hub mqtt certificate).
And details can contain the real error that may be an error from go http client.

File: scaleway/resource_iot_hub_test.go [hunk 11]
```diff
@@ -64,6 +64,9 @@ func TestAccScalewayIotHub_Minimal(t *testing.T) {
 					resource.TestCheckResourceAttrSet("scaleway_iot_hub.minimal", "endpoint"),
 					resource.TestCheckResourceAttr("scaleway_iot_hub.minimal", "device_count", "0"),
 					resource.TestCheckResourceAttr("scaleway_iot_hub.minimal", "connected_device_count", "0"),
+					// If the plan is shared, there is no CQTT CA, so it should be empty
```
Comment (remyleone) [inline scaleway/resource_iot_hub_test.go:67]:
```suggestion
					// If the plan is shared, there is no MQTT CA, so it should be empty
```

**Review by remyleone — APPROVED** [review]

**Pull request merged**